src/dso_api/dynamic_api/permissions.py

```diff
@@ -192,31 +180,36 @@ def _check_field_access(  # noqa: C901
             # Relations with non-composite keys become a pseudo-property with the target table's
             # identifier appended to the table name: other_table_id. After camel-casing, that
             # becomes OtherTableId. The identifier can be anything, but here we don't have enough
-            # information to cheaply determine what it was, so we assume it's "id", strip that off
-            # and retry. Note that docs/source/datasets.py also assumes "Id", so any other
-            # identifier already breaks the docs.
-            #
-            # TODO The dotted syntax for composite-key relations needs to be extended to this case.
+            # information to cheaply determine what it was, so we assume it's "id" and retry.
+            # Note that docs/source/datasets.py also assumes "Id", so any other identifier already
+            # breaks the docs.
             if part.endswith("Id"):
                 part = part[: -len("Id")]
-                field_name = part + ("." + field_name if field_name else "")
+                field_name = part + ".id" + ("." + field_name if field_name else "")
```

Comment:
@lbam vraagje, werkt dit wel? Want dit gaat bij de volgende loop denk ik mis?

Reply:
Nee, dit was stuk. Zit ook al niet meer in master.